src/App.tsx

```diff
@@ -14,11 +15,90 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+interface State {
+  selectedGoods: string[];
+}
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: [],
+  };
+
+  addToSelectedGood = (good: string) => {
+    this.setState((state) => ({
+      selectedGoods: [...state.selectedGoods, good],
+    }));
+  };
+
+  removeFromSelectedGood = (good: string) => {
+    this.setState((state) => ({
```

Comment:
тут аналогічно, назву краще змінити щоб не виникало проблем з неймінгом, бо змінна state вже оголошена

```suggestion
    this.setState((currentState) => ({
```

Reply:
fixed